### PR TITLE
run 01171 test in parallel

### PR DESCRIPTION
**Pull request by CheSema**
I added `MAX_ITERATION` and `MIN_TIME`, `MAX_TIME`.

Now the logic is folowed:
- run each thread not less that `MIN_TIME`
- not much that `MAX_ITERATION` and `MAX_TIME`
- after that it is checked that `MIN_ITERATION` has been achieved.

That allows to run fast on release build and up to `MAX_TIME` on slow builds.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh [hunk 1]
```diff
@@ -42,14 +36,14 @@ function run_until_deadline_and_at_least_times()
...
-    started_time=$(get_now)
+    started_time=$SECONDS
     local i=0
 
     while true
     do
         $function_to_run $i "$@"
 
-        [[ $(get_now) -lt $deadline ]] || break
+        [[ $SECONDS -lt $deadline ]] || break
```
Comment (Algunenano) [inline tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:46]: 400 seconds seem too much, no matter if in parallel or sequential. Why not break out if a certain  number of iterations is reached?

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 94aadc86610513d1b2f07938bbbd0e05aac932d9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67470/94aadc86610513d1b2f07938bbbd0e05aac932d9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/83701144004f8f3ccd8c8553cfcdbcd13e197231/docs_check.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/05f1d122b1c981a799e3293f2985df1f2a527212/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh (hunk 1 above)
Comment (CheSema) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:46]: > Why not break out if a certain number of iterations is reached?
It was like that recently. It required 100 iterations.
100 iteration on release take ~100 seconds. In debug it takes more that 10 minutes.
So we need to make this iteration limited by time.

I could add `MAX_ITERATION` that would be what you want.

Comment (Algunenano) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:47]: I see sub-10 second runs in master with release mode: https://s3.amazonaws.com/clickhouse-test-reports/0/8f60b9776fef6e14ca0e4b4a8b88ab329f17d3f2/stateless_tests__release_.html:
```
01171_mv_select_insert_isolation_long	OK	8.94
```

Comment (Algunenano) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:47]: In fact we can see that it has skyrocketed:

https://play.clickhouse.com/play?user=play#U2VsZWN0IHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBhcyB0LCBhdmcodGVzdF9kdXJhdGlvbl9tcykKZnJvbSBjaGVja3MKd2hlcmUgdGVzdF9uYW1lID0gJzAxMTcxX212X3NlbGVjdF9pbnNlcnRfaXNvbGF0aW9uX2xvbmcnIGFuZCB0ZXN0X3N0YXR1cz0nT0snIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCBhbmQgY2hlY2tfc3RhcnRfdGltZSA+ICcyMDI0LTA3LTAxIDAwOjAwOjAwJyBhbmQgY2hlY2tfbmFtZSA9ICdTdGF0ZWxlc3MgdGVzdHMgKHJlbGVhc2UpJwpHUk9VUCBieSB0Ck9SREVSIEJZIHQgQVND

It was 12 seconds until 10 of July, then it got 50%-100% slower around 10th of June, and now it's 20x slower

Comment (CheSema) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:47]: yes, you are right, it is fast in release build

https://play.clickhouse.com/play?user=play#c2VsZWN0IGNoZWNrX25hbWUsIGNvdW50KCkgYXMgYywgcXVhbnRpbGVzKDAuNSwgMC45LCAwLjk5LCAwLjk5OTkpKHRlc3RfZHVyYXRpb25fbXMpLCBtYXgodGVzdF9kdXJhdGlvbl9tcykgYXMgbWF4X2QgZnJvbSBjaGVja3Mgd2hlcmUgcHVsbF9yZXF1ZXN0X251bWJlciA9IDAgYW5kIGNoZWNrX3N0YXJ0X3RpbWUgPiAnMjAyNC0wMS0xMCAxODozMzoxMycgYW5kIHRlc3RfbmFtZSA9ICcwMTE3MV9tdl9zZWxlY3RfaW5zZXJ0X2lzb2xhdGlvbl9sb25nJyBhbmQgdGVzdF9zdGF0dXM9J09LJyBncm91cCBieSBjaGVja19uYW1lIGhhdmluZyBjID4gNTAgb3JkZXIgYnkgbWF4X2Q=

**Comment by CheSema** [comment]
Stateless tests flaky check (asan) 
01171_mv_select_insert_isolation_long -- 1 out of 30 has failed:
`19/20 : not enough iterations of select_insert_action has been made from 14 until 414`
I could lover setting a bit more, but I think it is ok.  No need to run it one more time.

**Review by Algunenano — COMMENTED** [review]
Kind of complex, but it looks fine. OTOH, the [test is flaky](https://s3.amazonaws.com/clickhouse-test-reports/67470/9994659541f0129a8fea96f7d6126088b62dc3ab/stateless_tests_flaky_check__asan_.html) so probably not fine.

**Comment by CheSema** [comment]
> Kind of complex, but it looks fine. OTOH, the [test is flaky](https://s3.amazonaws.com/clickhouse-test-reports/67470/9994659541f0129a8fea96f7d6126088b62dc3ab/stateless_tests_flaky_check__asan_.html) so probably not fine.

Ok, one more run.

**Review by Algunenano — APPROVED** [review]

**Comment by CheSema** [comment]
Stateless tests flaky check (asan) — ok

**Pull request merged**